google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/sdk/CloudSdkPanel.java

```diff
@@ -46,20 +46,10 @@
   private JLabel warningMessage;
   private JPanel cloudSdkPanel;
 
-  public CloudSdkPanel(@NotNull CloudSdkService cloudSdkService) {
+  public CloudSdkPanel() {
     warningMessage.setVisible(false);
     warningMessage.setIcon(RunConfigurations.ConfigurationWarning);
 
-    if (cloudSdkService.getSdkHomePath() == null) {
-      final String cloudSdkDirectoryPath
-          = CloudSdkUtil.findCloudSdkDirectoryPath(SystemEnvironmentProvider.getInstance());
```

Comment:
It happens inside the tools lib. See DefaultCloudSdkService.getSdkHomePath().

In this case, reset() is called, which checks if the CLOUD_SDK_PROPERTY_KEY is null. If it is, it uses CloudSdk.getSdkPath(), which uses the default PathResolver to find the gcloud binary.